code/mob/living/critter/ai/ocean.dm

```diff
@@ -65,7 +64,7 @@
 			var/list/possible = get_targets()
 			if (possible.len)
 				holder.target = pick(possible)
-	if(holder.target)
+	if(holder.target && holder.target.z == holder.owner.z)
 		var/mob/living/M = holder.target
 		if(!isalive(M))
```

Comment:
```suggestion
	if(holder.target)
		var/mob/living/M = holder.target
		if(!isalive(M) || !(M.z == holder.owner.z))
```

what about doing it like this so it clears the target and looks for a new one once you leave the Z?

Reply:
Considered that, but decided that letting the frustration check handle it would work instead, so if you quickly swapped Z levels back and forth it wouldn't drop aggro